kinesis/kinesis_test.go

```diff
@@ -204,3 +206,29 @@ func TestZlibCompressionEmpty(t *testing.T) {
 	_, err := zlibCompress(nil)
 	assert.NotEqual(t, err, nil, "'nil' data should return an error")
 }
+
+func TestDotReplace(t *testing.T) {
+	records := make([]*kinesis.PutRecordsRequestEntry, 0, 500)
+	record := map[interface{}]interface{}{
+		"testkey": []byte("test value"),
+		"kubernetes": map[interface{}]interface{}{
+			"app":                    []byte("test app label"),
+			"app.kubernetes.io/name": []byte("test key with dots"),
+		},
+	}
```

Comment:
I like this test case because the kubernetes bit is a real world example. 

Let's make it a bit more complicated though, just to be super safe. Add another top level key that has dots and make the value of that key be a map with some keys with dots too.

Reply:
Sure. Actually I have tested these cases with the dummy input by fluent bit. Will cover them in our unit test.